types/lodash/lodash-tests.ts

```diff
@@ -2809,6 +2811,8 @@ _.chain([1, 2, 3, 4]).unshift(5, 6); // $ExpectType LoDashExplicitWrapper<number
         value; // $ExpectType AbcObject
         return true;
     });
+    // $ExpectType [number[], string[]]
+    _.partition<number, string>(mixedArray, (value) => typeof value === 'number');
```

Comment:
If you use the type guard as suggested in my other comment, then you should be able to write this without explicitly specifying any generic types, e.g.

```ts
 _.partition(mixedArray, (value): value is number => typeof value === 'number');
```

or more simply:

```ts
 _.partition(mixedArray, _.isNumber);
```

Reply:
Updated and moved `partition` definition 
```
partition<T1, T2>(
  collection: List<T1 | T2> | null | undefined,
  callback: ValueIteratorTypeGuard<T1 | T2, T1>
): [T1[], T2[]];
partition<T>(
  collection: List<T> | null | undefined,
  callback: ValueIteratee<T>
): [T[], T[]];
```

```
_.partition(mixedArray, (value): value is number => typeof value === 'number');
```
> Expected type to be:
  [number[], string[]]
got:
  [number[], (string | number)[]]

TypeScript thinks that T2 is `number | string`

Reply:
Weird, I guess you'll need to keep the explicit `<number, string>` then.

Actually, another thing you could try:

```ts
partition<T, U extends T>(
    collection: List<T> | null | undefined,
    callback: ValueIteratorTypeGuard<T, U>
): [U[], Array<Exclude<T, U>>];
```

This should forcibly exlude `number` from the second array in the return value. It should also work better when converted into the `fp.d.ts` format. (Note that the generic types represent slightly different things now - `T` is `number|string` and `U` is `number`.)

Reply:
Looks like the alternative approach works!